examples/kitchen-sink.yaml

```diff
@@ -125,6 +125,8 @@ meter_provider:
             without_units: false
             without_type_suffix: false
             without_scope_info: false
+            # Configure Prometheus Exporter to add resource attributes as metrics attributes.
+            with_resource_as_constant_labels: "service\\.[.]*"
```

Comment:
In #64 I proposed using the view instrument name wildcard syntax, where `?` matches any single character, and `*` matches any number of characters including none. 

The argument being that this avoids having to answer questions about which flavor of regex is used, and whether we can actually accommodate that across languages.

The downside of a simpler wildcard syntax is that you likely need to switch this parameter from a `string` type to an array of strings, since you can easily run into situations where you can't express the desired set in a single expression.

Reply:
Right, another downside is with a wildcard there isn't a mechanism for exclusions built-in

Reply:
What do you think of #71?

Reply:
The benefit of #71 is that we get ahead of what I think will be a recurring pattern of wanting to provide a way in file config to specify an allow list or disallow list with pattern matching. Regex comes with the complexity of specifying the version of regular expressions, and if a language doesn't support the specification, we'll have worse interoperability because regexes will need to be adjusted for an individual language's requirements. Glob pattern matching is straight forward to implement and gets most of there of what you want to do with regex.

Reply:
What are your thoughts on how one would use #71 to include everything except something that matches? I'm thinking specifically in the collector where someone may want to include all resource attributes except for maybe one of them.

Reply:
I like the simplicity of wildcards better, so long as we retain the ability to have an include or exclude match, then that would solve my use-case

Reply:
Various components in the collector support the following for include/exclude rules:

```yaml
include:
    - /var/log/pods/default_*/*/*.log
exclude: []
```

@jack-berg if you're ok with this, it can be added to #71

Reply:
Yeah that sounds great. Do you want to reflect that in this PR and merge? Then I can update #71 in a followup to reflect that syntax we use here.